lib/srv6.h

```diff
@@ -167,12 +175,20 @@ static inline bool sid_diff(
 	return !sid_same(a, b);
 }
 
-static inline bool sid_zero(
-		const struct in6_addr *a)
+
+static inline bool sid_zero(const struct seg6_seg_stack *a)
+{
+	struct in6_addr zero = {};
+	if (!a)
+		return true;
```

Comment:
false right? any reason for returning true?

Reply:
Hello @routingrocks , Thank you for your comment.

This function returns 'true' if `seg6_seg_stack`  is a NULL pointer.

- In the next context we can free `nexthop->nh_srv6 only` if seg6_seg_stack if NULL, so function should return 'true': https://github.com/FRRouting/frr/blob/master/lib/nexthop.c#L573
- In the next context we can't enter the nexthop_add_srv6_seg6() in which seg6_segs is assigned, but not allocated, because function should return 'true'  when seg6_seg is not allocated. So condition 'if-branch' is not entered. https://github.com/FRRouting/frr/blob/master/lib/nexthop.c#L813
- In the next context we don't enter the if statement where we take data from seg6_segs that is not allocated https://github.com/FRRouting/frr/blob/master/zebra/rt_netlink.c#L1628
And other examples:
- https://github.com/FRRouting/frr/blob/master/lib/zclient.c#L2002
- https://github.com/FRRouting/frr/blob/master/lib/zclient.c#L2063

Considering the contexts, I'm okay with that this return `true` in this case.
Please let me know if it is okay for you.
Thanks!

Reply:
to further @routingrocks point-> People are going to assume a sid_zero returns true/false associated with a sid of zero or not.   A null pointer typically does not mean something is true( quite the opposite in fact!  )   hence the question.  if a null pointer is important to be true then *please* *please* document this at the top of the function and where it is defined in the header file as well.  Future developers will see this and question what the hell is going on.